x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/DataTypeConversion.java

```diff
@@ -470,13 +549,21 @@ public static Object convert(Object value, DataType dataType) {
         private static Function<Object, Object> fromBool(Function<Boolean, Object> converter) {
             return (Object l) -> converter.apply(((Boolean) l));
         }
-        
-        private static Function<Object, Object> fromDate(Function<Long, Object> converter) {
-            return l -> ((ZonedDateTime) l).toEpochSecond();
+
+        private static Function<Object, Object> fromDateTime(Function<Long, Object> converter) {
+            return l -> converter.apply(((ZonedDateTime) l).toEpochSecond());
```

Comment:
Shouldn't it be epoch millis?

Reply:
I added a new issue #37655 and have a separate PR because we want to backport this fix to earlier versions.